tcf_website/static/department/loadMoreCourses.js

```diff
@@ -0,0 +1,86 @@
+async function loadPage(subdepartmentId, url) {
+    if(url === null) {
+        hideSpinner(subdepartmentId);
+        return;
+    };
+    const courses = await fetch(url).then(res => res.json());
+    if(courses.detail === "Invalid page."){
+        hideSpinner(subdepartmentId);
+        return;
+    }
+    else{
+        courses.results.forEach((course) => {
+            let element = document.getElementById(`courses-sd-${subdepartmentId}`);
+            const html = generateCourseCardHTML(course);
+            element.insertAdjacentHTML('beforeend', html);
+        });
+        loadPage(subdepartmentId, courses.next);
+    }
+}
+
+function generateCourseCardHTML(course) {
+    const rating = emdashOrTwoDecimals(course.average_rating);
+    const difficulty = emdashOrTwoDecimals(course.average_difficulty);
+    const seasonAllCaps = course.semester_last_taught.season;
+    const seasonTitleCase = seasonAllCaps.charAt(0) + seasonAllCaps.substr(1).toLowerCase();
+    return `

```

Comment:
Is it possible to move this HTML to a template and just pass in param values you need? Would prefer to keep presentation separate from functionality.

Reply:
I've thought about it and did some research, but I can't think of a way to do that. I'll ask someone with more experience.

Reply:
Didn't get a helpful response, and I can't think of a better way. I guess we can refactor this later?